docs/database/mongodb-integration.md

```diff
@@ -1,137 +1,310 @@
 
-## Example usage
+### Add MongoDB server resource and database resource
 
-In the _:::no-loc text="Program.cs":::_ file of your client-consuming project, call the `AddMongoDBClient` extension to register a `IMongoClient` for use via the dependency injection container.
+In your app host project, call <xref:Aspire.Hosting.MongoDBBuilderExtensions.AddMongoDB*> to add and return a MongoDB server resource builder. Chain a call to the returned resource builder to <xref:Aspire.Hosting.MongoDBBuilderExtensions.AddDatabase*>, to add a MongoDB database resource.
 
 ```csharp
-builder.AddMongoDBClient("mongodb");
+var builder = DistributedApplication.CreateBuilder(args);
+
+var mongo = builder.AddMongoDB("mongo")
+                   .WithLifetime(ContainerLifetime.Persistent);
+
+var mongodb = mongo.AddDatabase("mongodb");
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(mongodb)
+       .WaitFor(mongodb);
+
+// After adding all resources, run the app...
 ```
 
-To retrieve your `IMongoClient` object, consider the following example service:
+> [!NOTE]
+> The MongoDB container can be slow to start, so it's best to use a _persistent_ lifetime to avoid unnecessary restarts. For more information, see [Container resource lifetime](../fundamentals/app-host-overview.md#container-resource-lifetime).
 
-```csharp
-public class ExampleService(IMongoClient mongoClient)
+When .NET Aspire adds a container image to the app host, as shown in the preceding example with the `docker.io/library/mongo` image, it creates a new MongoDB instance on your local machine. A reference to your MongoDB server resource builder (the `mongo` variable) is used to add a database. The database is named `mongodb` and then added to the `ExampleProject`. The MongoDB server resource includes default credentials:
+
+- `MONGO_INITDB_ROOT_USERNAME`: A value of `admin`.
+- `MONGO_INITDB_ROOT_PASSWORD`: Random `password` generated using the <xref:Aspire.Hosting.ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter*> method.
+
+When the app host runs, the password is stored in the app host's secret store. It's added to the `Parameters` section, for example:
+
+```json
 {
-    // Use mongoClient...
+  "Parameters:mongo-password": "<THE_GENERATED_PASSWORD>"
 }
 ```
 
-After adding a `IMongoClient`, you can require the `IMongoClient` instance using DI.
+The name of the parameter is `mongo-password`, but really it's just formatting the resource name with a `-password` suffix. For more information, see [Safe storage of app secrets in development in ASP.NET Core](/aspnet/core/security/app-secrets) and [Add MongoDB server resource with parameters](#add-mongodb-server-resource-with-parameters).
+
+The <xref:Aspire.Hosting.ResourceBuilderExtensions.WithReference%2A> method configures a connection in the `ExampleProject` named `mongodb`.
```

Comment:
It would be good to also describe what `.WaitFor` does here since you are describing what `.WithReference` does.